spec/rubocop/cop/performance/count_spec.rb

```diff
@@ -177,101 +177,139 @@ def count(&block)
+  context 'with `reject` and `size`' do
+    it 'registers an offense' do
+      expect_offense(<<~RUBY)
+        array.reject { |e| e > 2 }.size
+              ^^^^^^^^^^^^^^^^^^^^^^^^^ Use `count` instead of `reject...size`.
+      RUBY
 
-        expect_no_corrections
-      end
+      expect_correction(<<~RUBY)
+        array.count { |e| !(e > 2) }
+      RUBY
+    end
+  end
 
-      it 'reject...length' do
-        expect_offense(<<~RUBY)
-          [1, 2].reject { |e| e > 2 }.length
-                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `count` instead of `reject...length`.
-        RUBY
+  context 'with `reject` and `count`' do
+    it 'registers an offense' do
+      expect_offense(<<~RUBY)
+        array.reject { |e| e > 2 }.count
+              ^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `count` instead of `reject...count`.
+      RUBY
 
-        expect_no_corrections
-      end
+      expect_correction(<<~RUBY)
+        array.count { |e| !(e > 2) }
+      RUBY
+    end
+  end
 
-      it 'select...count when count has a block' do
-        expect_no_offenses(<<~RUBY)
-          [1, 2].select { |e| e > 2 }.count { |e| e.even? }
-        RUBY
-      end
+  context 'with `reject` and `length`' do
+    it 'registers an offense' do
+      expect_offense(<<~RUBY)
+        array.reject { |e| e > 2 }.length
+              ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `count` instead of `reject...length`.
+      RUBY
 
-      it 'reject...size when select has parameters' do
-        expect_offense(<<~RUBY)
-          Data = Struct.new(:value)
-          array = [Data.new(2), Data.new(3), Data.new(2)]
-          puts array.reject(&:value).size
-                     ^^^^^^^^^^^^^^^^^^^^ Use `count` instead of `reject...size`.
-        RUBY
+      expect_correction(<<~RUBY)
+        array.count { |e| !(e > 2) }
+      RUBY
+    end
+  end
 
-        expect_no_corrections
-      end
+  context 'with `reject` with symbol block argument and `size`' do
+    it 'registers an offense' do
+      expect_offense(<<~RUBY)
+        array.reject(&:value).size
+              ^^^^^^^^^^^^^^^^^^^^ Use `count` instead of `reject...size`.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        array.count { |element| !element.value }
+      RUBY
+    end
+  end
+
+  context 'with `reject` with variable block argument and `size`' do
+    it 'registers an offense' do
+      expect_offense(<<~RUBY)
+        array.reject(&block).size
+              ^^^^^^^^^^^^^^^^^^^ Use `count` instead of `reject...size`.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        array.count { !block.call }
+      RUBY
+    end
+  end
+
+  context 'with `reject` with some statements and `length`' do
+    it 'registers an offense' do
+      expect_offense(<<~RUBY)
+        array.reject {
+              ^^^^^^^^ Use `count` instead of `reject...length`.
+          foo
+          bar
+        }.length
+      RUBY
+
+      expect_correction(<<~RUBY)
+        array.count {
+          !(foo
+          bar)
+        }
```

Comment:
It looks like it needs to be auto-corrected to the following to be compatible.

```suggestion
        array.count {
          foo
          !bar
        }
```

Reply:
I think that is the point I mentioned at the end of the description, do you have any better ideas to achieve that, and what about this thought?

>other cops autocorrects it to some other good form

By the way, you say "for compatible", what exactly is incompatible?

Reply:
Ah..., it doesn't contribute to compatibility. But `!` to `foo` makes no sense.

Reply:
Perhaps the implementation that Solargraph uses to estimate the method's return type might be helpful:

- https://github.com/castwide/solargraph/blob/12fcf71755db2d570f591de773753eb73ac0680c/lib/solargraph/parser/legacy/node_methods.rb#L226-L256

If we have a similar implementation, we can see which nodes can be the return value of a block, and if we negate them, I think we can achieve most of what we want to do. Though this may be a little too heavy implementation for this Cop to handle.

Reply:
I think it can be achieved by inverting the last sentence of block with `!`. Anyway, let's make a better autocorrect separately.

Reply:
I changed it so that the only last statement is negated in the case of `begin` node as a simple enough approach. I think indeed it's going to work there, even if it's not a perfect autocorrection.